tendrl/node_agent/flows/import_cluster/__init__.py

```diff
@@ -42,3 +44,16 @@ def run(self):
             import_ceph(tendrl_ns.tendrl_context.integration_id)
         else:
             import_gluster(tendrl_ns.tendrl_context.integration_id)
+
+        # import cluster's run() should not return unless the new cluster entry
+        # is updated in etcd, as the job is marked as finished if this
+        # function is returned. This might lead to inconsistancy in the API
+        # functionality. The below loop waits for the cluster details
+        # to be updated in etcd.
+        while True:
+            gevent.sleep(2)
+            try:
+                tendrl_ns.etcd_orm.client.read("/clusters/%s" % integration_id)
+                break
+            except etcd.EtcdKeyNotFound:
+                continue
```

Comment:
How do we mark a task as failed if some exception etc happens while execution of the flows? I feel the same should be handled better way in commons job handler. Can you create an issue and send a PR for the same?